Assets/Scripts/EntityBehaviour.cs

```diff
@@ -39,21 +51,59 @@ public virtual void Death()
         //ADD COMBO
         FindObjectOfType<ComboBehaviour>().TriggerAddCombo(1);
 
+        
+
         //CHECK FOR END OF COMBAT - Spawn pick ups at end of carriage
         if (FindObjectOfType<CombatManager>().CheckForEndCombat(gameObject))
         {
             GameObject _prefab = Resources.Load<GameObject>("TemporaryPickUp");
+            //Spawn TNT pick up at 100% at Furnace room
             if (FindObjectOfType<CombatManager>().currentCarriage._isSpecialCarriage == CarriageData.SpecialCarriageExceptions.Furnace)
+            {
                 _prefab.GetComponent<TemporaryPickUp>().isTnt = true;
-            else _prefab.GetComponent<TemporaryPickUp>().isTnt = false;
-            Instantiate(_prefab, transform.position, Quaternion.identity);
+                Instantiate(_prefab, transform.position, Quaternion.identity);
+            }               
+
+            //Spawn HP pick up at 100% at end of each other room if player is low health
+            else if (FindObjectOfType<CharacterBehaviour>().isLowHealth)
+            {
+                _prefab.GetComponent<TemporaryPickUp>().isTnt = false;
+                Instantiate(_prefab, transform.position, Quaternion.identity);
+            }
+
+            //Spawn HP pick up at 40% at end of each other room
+            else
+            {
+                int _RNGesus = Random.Range(0, 101);
+                if (_RNGesus >= 60)
+                {
+                    Debug.Log("RNG: " + _RNGesus + ". Report if RNG is below 60.");
+                    _prefab.GetComponent<TemporaryPickUp>().isTnt = false;
+                    Instantiate(_prefab, transform.position, Quaternion.identity);
+                }               
+            }
         }
 
-        //CHANCE TO DROP BULLET
+        //CHANCE TO DROP BULLET - Only in the middle of combat
         else
         {
-            GameObject _prefab = Resources.Load<GameObject>("BulletPickUp");
-            Instantiate(_prefab, transform.position, Quaternion.identity);
+            //Roll 1-100s
+            int _RNGesus = Random.Range(0, 101);
+            if (_RNGesus >= 75)
```

Comment:
There are 101 possible values here, not 100. This is a fencepost problem I think, your probabilities will be off by 1ish percent.

Consider using this: https://docs.unity3d.com/ScriptReference/Random-value.html

`if (Random.value > 0.75) [...]` would work perfectly well i think

Reply:
changed to random.value. This was a great suggestion, thanks for introducing this to me.